src/metabase/driver/postgres.clj

```diff
@@ -209,6 +213,58 @@
 
 (def ^:private ^:dynamic *enum-types* nil)
 
+(defn- get-tables-sql
+  [schemas table-names]
+  ;; Ref: https://github.com/davecramer/pgjdbc/blob/a714bfd/pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java#L1272
+  (sql/format
+   (cond-> {:select    [[:n.nspname :schema]
+                        [:c.relname :name]
+                        [[:case-expr :c.relkind
+                          [:inline "r"] [:inline "TABLE"]
+                          [:inline "p"] [:inline "PARTITIONED TABLE"]
+                          [:inline "v"] [:inline "VIEW"]
+                          [:inline "f"] [:inline "FOREIGN TABLE"]
+                          [:inline "m"] [:inline "MATERIALIZED VIEW"]
+                          :else nil]
+                         :type]
+                        [:d.description :description]]
+            :from      [[:pg_catalog.pg_namespace :n]
+                        [:pg_catalog.pg_class :c]]
+            :left-join [[:pg_catalog.pg_description :d] [:and [:= :c.oid :d.objoid] [:= :d.objsubid [:inline 0]]]
+                        [:pg_catalog.pg_class :dc]      [:and [:= :d.classoid :dc.oid] [:= :dc.relname [:inline "pg_class"]]]
+                        [:pg_catalog.pg_namespace :dn]  [:and [:= :dn.oid :dc.relnamespace] [:= :dn.nspname [:inline "pg_catalog"]]]]
+            :where     [:and [:= :c.relnamespace :n.oid]
+                        ;; filter out system tables
+                        [(keyword "!~") :n.nspname "^pg_"] [:<> :n.nspname "information_schema"]
+                        ;; only get tables of type: TABLE, PARTITIONED TABLE, VIEW, FOREIGN TABLE, MATERIALIZED VIEW
+                        [:raw "c.relkind in ('r', 'p', 'v', 'f', 'm')"]]
+            :order-by  [:type :schema :name]}
+     (seq schemas)
+     (sql.helpers/where [:in :n.nspname schemas])
+
+     (seq table-names)
+     (sql.helpers/where [:in :c.relname table-names]))
+   {:dialect :ansi}))
+
+(defn- get-tables
+  ;; have it as its own method for ease of testing
```

Comment:
Good idea leaving this comment here. Will help anyone thinking of "cleaning it up" in the future not to try it 👍